presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -104,10 +107,9 @@
     protected final String connectionUrl;
     protected final Properties connectionProperties;
     protected final String identifierQuote;
-    private final boolean mapTableNames;
 
-    private Map<String, Map<String, String>> schemaTableMapping = new HashMap<>();
-    private Map<String, ReentrantLock> schemaTableMappingLock = new ConcurrentHashMap<>();
+    private final LoadingCache<String, Optional<String>> schemaMappingCache;
+    private final Map<String, LoadingCache<String, Optional<String>>> schemaTableMapping = new ConcurrentHashMap<>();
```

Comment:
I would hide these fields and related methods as a separate class. Then it seems that with some abstraction over JDBC (which is used to load the cache) you could write some additional low level unit tests without using mysql. What do you think?

Reply:
I see. Sounds a bit like an over engineering.